lib/model.js

```diff
@@ -1134,9 +1151,14 @@ Connection.from = (url, callback) => {
         // authSource takes precedence, but fall back to admin
         // @note Durran: This is not the documented behaviour of the connection string
         // but the shell also does not fall back to the dbName and will use admin.
-        attrs.mongodbDatabaseName = decodeURIComponent(
-          attrs.authSource || Connection.MONGODB_DATABASE_NAME_DEFAULT
-        );
+        try {
+          attrs.mongodbDatabaseName = decodeURIComponent(
+            attrs.authSource || Connection.MONGODB_DATABASE_NAME_DEFAULT,
+            callback
```

Comment:
Hmmm what does the callback do here? I think might be an extra? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/decodeURIComponent